src/core/storage/sframe_interface/unity_sarray.cpp

```diff
@@ -1102,6 +1102,59 @@ flexible_type unity_sarray::mean() {
   }
 }
 
+flexible_type unity_sarray::median(bool approx) {
+  log_func_entry();
+
+  flex_type_enum type = dtype();
+  if(type != flex_type_enum::INTEGER && type != flex_type_enum::FLOAT) {
+    log_and_throw("Can not calculate median on non-numeric SArray.");
+  }
+  if (size() == 0) {
+    return flex_undefined();
+  }
+
+  // Use sketch to get a fast approximate answer
+  turi::unity_sketch* sketch = new turi::unity_sketch();
+  gl_sarray data = std::static_pointer_cast<unity_sarray>(shared_from_this());
+  data = data.dropna();
+  sketch->construct_from_sarray(data);
+  double approx_median = sketch->get_quantile(0.5);
+
+  flexible_type result;
+  if (!approx) {
+    const double epsilon = sketch->numeric_epsilon();
+    const double upper_bound = approx_median + (epsilon * data.size());
+    const double lower_bound = approx_median - (epsilon * data.size());
```

Comment:
Unfortunately, the guarantees on the quantile don't work by value, but by element count.  The approximate median returned is the element that is in position +/- eps*data.size() of the true median.  Thus to get the upper and lower bounds, you need to actually do: 
```
const double upper_bound = sketch->get_quantile(0.5 + epsilon * data.size()); 
const double lower_bound = sketch->get_quantile(0.5 - epsilon * data.size()); 
```